specs/core/1_custody-game.md

```diff
@@ -288,32 +329,93 @@ Add the following operations to the per-block processing, in order the given bel
 
 #### Custody reveals
 
-Verify that `len(block.body.custody_key_reveals) <= MAX_CUSTODY_KEY_REVEALS`.
-
-For each `reveal` in `block.body.custody_key_reveals`, run the following function:
+Replace process_custody_reveal from phase 0 with this function:
 
 ```python
 def process_custody_reveal(state: BeaconState,
-                           reveal: CustodyKeyReveal) -> None:
+                           reveal: RandaoKeyReveal) -> None:
+
+    revealed_validator = state.validator_registry[reveal.revealed_index]
+
+    assert reveal.epoch == get_randao_epoch_for_custody_period( revealed_validator.next_custody_reveal_period, reveal.revealed_index)
+    assert revealed_validator.next_custody_reveal_period < get_validators_custody_reveal_period(state, reveal.revealed_index)
+
+    # Revealed validator is active or exited, but not withdrawn
+    assert is_slashable_validator(revealer, get_current_epoch(state))
+
+    # Decrement max reveal lateness if response is timely
+    if revealed_validator.next_custody_reveal_period == get_validators_custody_reveal_period(state, reveal.revealer_index) - 2:
+            revealer.max_reveal_lateness -=  MAX_REVEAL_LATENESS_DECREMENT
+    revealed_validator.max_reveal_lateness = max(revealed_validator.max_reveal_lateness, get_validators_custody_reveal_period(state, reveal.revealed_index) - revealed_validator.next_custody_reveal_period)
+
+    # Process reveal
+    revealed_validator.next_custody_reveal_period += 1
+
+    # Reward Block Preposer
+    proposer_index = get_beacon_proposer_index(state)
+    increase_balance(state, proposer_index, base_reward(state, index) // MINOR_REWARD_QUOTIENT)
+```
+
+##### Randao key reveals
+
+Verify that `len(block.body.randao_key_reveals) <= MAX_RANDAO_KEY_REVEALS`.
+
+For each `reveal` in `block.body.randao_key_reveal`, run the following function:
+
+```python
+def process_randao_key_reveal(state: BeaconState,
+                              reveal: RandaoKeyReveal) -> None:
+    """
+    Process ``RandaoKeyReveal`` operation.
+    Note that this function mutates ``state``.
+    """
+    # Verify signature correctness
     assert verify_custody_key(state, reveal)
```

Comment:
Maybe call it something other than `verify_custody_key`? Like `verify_randao_reveal` or something similar.

Reply:
Removed the function now as it seemed pointless